### PR TITLE
OBPIH-5433 Add Validation for Stock Movement deletion (fix after QA)

### PR DESCRIPTION
**Pull request by alannadolny**

### CODE REVIEW

**Review by drodzewicz — APPROVED** [review]

File: grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy [hunk 1]
```diff
@@ -176,7 +176,7 @@ class StockMovementApiController {
             try {
                 stockTransferService.deleteStockTransfer(params.id)
             } catch (Exception e) {
-                def message = "Unable to delete stock movement with ID ${params.id}: " + e.message
+                def message = "${warehouse.message(code: 'stockMovement.delete.error.message')}"
```
Comment (kchelstowski) [inline grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy:179]: We want to get rid of `warehouse.message` at some points, so let's use `g.message` if possible. Consider adding a default message also.
```suggestion
                def message = "${g.message(code: 'stockMovement.delete.error.message')}"
```

File: grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy [hunk 2]
```diff
@@ -195,7 +195,7 @@ class StockMovementApiController {
                     try {
                         stockMovementService.deleteStockMovement(params.id)
                     } catch (Exception e) {
-                        def message = "Unable to delete stock movement with ID ${params.id}: " + e.message
+                        def message = "${warehouse.message(code: 'stockMovement.delete.error.message')}"
```
Comment (kchelstowski) [inline grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy:198]: same as above

File: grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy [hunk 3]
```diff
@@ -213,7 +213,7 @@ class StockMovementController {
                     )
                 } catch (Exception e) {
                     log.error("Unable to delete stock movement with ID ${params.id}: " + e.message, e)
-                    flash.message = "Unable to delete stock movement with ID ${params.id}: " + e.message
+                    flash.message = "${warehouse.message(code: 'stockMovement.delete.error.message')}"
```
Comment (kchelstowski) [inline grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy:216]: same as above

**Review by kchelstowski — CHANGES_REQUESTED** [review]

File: grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy [hunk 4]
```diff
@@ -213,7 +213,7 @@ class StockMovementController {
                     )
                 } catch (Exception e) {
                     log.error("Unable to delete stock movement with ID ${params.id}: " + e.message, e)
-                    flash.message = "Unable to delete stock movement with ID ${params.id}: " + e.message
+                    flash.message = "${g.message(code: 'stockMovement.delete.error.message', body: 'The Stock Movement could not be deleted')}"
```
Comment (kchelstowski) [inline grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy:216]: Why are you using `body`?
```suggestion
                    flash.message = "${g.message(code: 'stockMovement.delete.error.message', default: 'The Stock Movement could not be deleted')}"
```

**Review by kchelstowski — CHANGES_REQUESTED** [review]

**Review by kchelstowski — APPROVED** [review]

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**